### PR TITLE
Report marker diagnostics during parsing, rather than evaluation

### PR DESCRIPTION
**Pull request by charliermarsh**
## Summary

I want to move towards a more normalized marker representation within the marker tree, which means that the things we warn against will disappear by the time we get to evaluation. I think it makes more sense to show these warnings when we create the tree, rather than when we evaluate it.

### CODE REVIEW

**Comment by charliermarsh** [comment]
Ah shoot, I accidentally merged https://github.com/astral-sh/uv/pull/9341 into this branch.

**Comment by charliermarsh** [comment]
Please review the first commit, but ignore the second.

File: crates/uv-pep508/src/marker/parse.rs [hunk 1]
```diff
@@ -101,14 +102,60 @@ pub(crate) fn parse_marker_value<T: Pep508Url>(
...
+            MarkerValue::from_str(key)
+                .map_err(|_| Pep508Error {
+                    message: Pep508ErrorSource::String(format!(
+                        "Expected a quoted string or a valid marker name, found `{key}`"
+                    )),
+                    start,
+                    len,
+                    input: cursor.to_string(),
+                })
+                .inspect(|value| match value {
```
Comment (konstin) [inline crates/uv-pep508/src/marker/parse.rs:114]: nit: `?` to remove the inspect

File: crates/uv-pep508/src/verbatim_url.rs [hunk 2]
```diff
@@ -9,8 +9,6 @@ use std::sync::LazyLock;
 use thiserror::Error;
 use url::{ParseError, Url};
 
-use uv_fs::{normalize_absolute_path, normalize_url_path};
```
Comment (konstin) [inline crates/uv-pep508/src/verbatim_url.rs:12]: I forgot to comment that, but this one is an explicit import to make exporting and rewriting to https://github.com/konstin/pep508_rs/blob/d155787317425e8ab897beef55a947e8ef28f9d8/src/verbatim_url.rs#L12-L13 easier

**Review by konstin — APPROVED** [review]
That's the much better place

File: crates/uv-pep508/src/marker/parse.rs (hunk 1 above)
Comment (charliermarsh) [reply crates/uv-pep508/src/marker/parse.rs:114]: Is that better? Then I have to assign and return the value

**Pull request merged**